### PR TITLE
#70 api backoffice

### PR DESCRIPTION
**Pull request by Myst3k69**

### CODE REVIEW

File: api/routes/backoffice/index.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
+const { Router } = require('express');
+const db = require('../../datasource/mysql');
+const router = Router();
+
+/* ********************* Routes for users handling ********************* */
+
+/* GET all users. */
+router.get('/users/all', (req, res) => {
+
+  db.query('SELECT * FROM user', (err, result) => {
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/index.js]: SELECT * :poop:

File: api/routes/backoffice/index.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      } else {
+
+// L'utilisateur a bien été trouvé dans la base de données
+        if (result[0].count > 0) {
+
+          // On renvoie les informations de l'utilisateur
+          db.query('SELECT name,mail,ip_adress,role_id FROM user WHERE id= ?', idUser, (err, result) => {
+
+            if (err) {
+              ;
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/index.js]: :poop:

File: api/routes/backoffice/index.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+});
+
+/* add one user */
+
+router.post('/users/add', (req, res) => {
+
+  const { name, mail, role } = req.body;
+
+  const pass = 'faqmdp';
+  const ip_adress = '127.0.0.1';
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/index.js]: const ipAddress = req.headers['x-forwarded-for'] || req.connection.remoteAddress;

File: api/routes/backoffice/user.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
+const { Router } = require('express');
+const db = require('../../datasource/mysql');
+const router = Router();
+
+/* ********************* Routes for users handling ********************* */
+
+/* GET all users. */
+router.get('/all', (req, res) => {
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/user.js]: get('/' :poop:

File: api/routes/backoffice/user.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        .send('User deleted');
+    }
+
+  });
+
+});
+
+/* add one user */
+
+router.post('/add', (req, res) => {
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/user.js]: post('/' :poop:

File: api/routes/backoffice/user.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        .send('User added');
+    }
+
+  });
+
+});
+
+/* UPDATE one user */
+
+router.put('/update/:id', (req, res) => {
```
Comment (maxencemeloniWCS) [inline api/routes/backoffice/user.js]: :poop:

**Review by ghost — APPROVED** [review]

**Pull request merged**